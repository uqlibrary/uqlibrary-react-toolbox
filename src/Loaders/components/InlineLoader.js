import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from 'material-ui/CircularProgress';

export default function InlineLoader({message}) {
    return (
        <div className="column align-center">
            <br /><br />
            <CircularProgress size={60} thickness={8} /> <br /><br />
            <span className="title is-4">{message}</span> <br /><br />
        </div>
    );
}

InlineLoader.propTypes = {
    message: PropTypes.string
};

InlineLoader.defaultProps = {
    message: 'Loading...'
};
