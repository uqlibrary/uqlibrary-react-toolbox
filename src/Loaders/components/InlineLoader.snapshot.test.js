jest.dontMock('./InlineLoader');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import InlineLoader from './InlineLoader';

function setup(message) {
    const props = {message};
    return shallow(<InlineLoader {...props} />);
}

describe('InlineLoader snapshots tests', () => {
    it('renders loader', () => {
        const wrapper = setup('Waiting to load...');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});