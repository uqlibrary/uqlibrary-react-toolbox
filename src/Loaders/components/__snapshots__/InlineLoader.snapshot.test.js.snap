// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLoader snapshots tests renders loader 1`] = `
<div
  className="column align-center"
>
  <br />
  <br />
  <CircularProgress
    max={100}
    min={0}
    mode="indeterminate"
    size={60}
    thickness={8}
    value={0}
  />
   
  <br />
  <br />
  <span
    className="headline"
  >
    Waiting to load...
  </span>
   
  <br />
  <br />
</div>
`;
